buildSrc/src/main/java/org/springframework/boot/build/ConventionsPlugin.java

```diff
@@ -141,6 +152,36 @@ private void applyJavaConventions(Project project) {
 		});
 	}
 
+	private void copyLegalFiles(Project project, CopySpec metaInf) {
+		copyNoticeFile(project, metaInf);
+		copyLicenseFile(project, metaInf);
+	}
+
+	private void copyNoticeFile(Project project, CopySpec metaInf) {
+		try {
+			InputStream notice = getClass().getClassLoader().getResourceAsStream("NOTICE.txt");
+			String noticeContent = FileCopyUtils.copyToString(new InputStreamReader(notice, StandardCharsets.UTF_8))
+					.replace("${version}", project.getVersion().toString())
+					.replace("${currentYear}", Integer.toString(Year.now().getValue()));
+			metaInf.from(project.getResources().getText().fromString(noticeContent),
```

Comment:
I think this'll break up-to-date checking for the jar. Creating a text resource from a string results in the creation of a temporary file with a random name. As that random name varies from build-to-build, the jar tasks will never be considered up-to-date as it'll look like its inputs have changed.

We had this problem when adding the classpath index support to `BootJar`. The solution was to get the text resource's file and rename it before adding it to the copy spec rather than as part of the spec.

Reply:
I will take a look how it was done there and report back

Reply:
Thanks. I should have linked to the relevant code earlier. It is the following:

https://github.com/spring-projects/spring-boot/blob/16111f126e707fb5064bab6150df7a77f54850ee/spring-boot-project/spring-boot-tools/spring-boot-gradle-plugin/src/main/java/org/springframework/boot/gradle/tasks/bundling/BootJar.java#L113-L119

This will result in a `NOTICE.txt` and `LICENSE.txt` file beneath each project's `build` directory. An alternative could be to access the files in `buildSrc/src/main/resources` directly from the filesystem.

Reply:
I tried that and that didn't work. Will take a look again.

Reply:
After the latest changes I get the following:
```
 ./gradlew :spring-boot-project:spring-boot-properties-migrator:jar

BUILD SUCCESSFUL in 1s
9 actionable tasks: 9 up-to-date
```

Previously, I got the following:
```
 ./gradlew :spring-boot-project:spring-boot-properties-migrator:jar

BUILD SUCCESSFUL in 1s
9 actionable tasks: 2 executed, 7 up-to-date
```

Thanks for the good catch, @wilkinsona